File: BlaBlaCar.BL/ODT/UserModel.cs
using BlaBlaCar.BL.ODT.CarModels;
using BlaBlaCar.BL.ODT.TripModels;
using Microsoft.Extensions.FileProviders.Physical;

namespace BlaBlaCar.BL.ODT
{
    public class UserModel
    {
        public string Id { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public string? FirstName { get; set; }
        public ModelStatus UserStatus { get; set; }
        public ICollection<UserDocumentsModel> UserDocuments { get; set; }
        public ICollection<CarModel> Cars { get; set; }
        public ICollection<TripModel> Trips { get; set; }
        public ICollection<TripUserModel> TripUsers { get; set; }
    }
}

File: web-client/src/app/interfaces/user-interfaces/user-model.ts
import { CarModel } from "../car-interfaces/car-model";
import { RoleModel } from "../role-model";
import { TripModel } from "../trip-interfaces/trip-model";
import { UserDocumentsModel } from "./user-documents-model";
import { UserStatus } from "./user-status";

export interface UserModel {
    id: string;
    email: string;
    firstName: string;
    phoneNumber: string;
    userImg?: string
    roles: any[];
    userDocuments: UserDocumentsModel[];
    userStatus: UserStatus;
    cars: CarModel[];
    trips?: TripModel[];
    tripUsers?: TripModel[];
    createdAt?: Date;

}
